assignments/lambda-classes.js

```diff
@@ -1 +1,130 @@
 // CODE here for your Lambda Classes
+
+class Person {
+    constructor(props) {
+        this.name = props.name;
+        this.age = props.age;
+        this.location = props.location;
+        this.gender = props.gender; 
+    }
+    speak () {
+        return `Hello, my name is ${this.name}, I am from ${this.location}.`
+    }
+} // end of Person
+
+class Instructor extends Person {
+    constructor(instructorProps) {
+        super(instructorProps);
+        this.specialty = instructorProps.specialty;
+        this.favLanguage = instructorProps.favLanguage;
+        this.catchPhrase = instructorProps.catchPhrase;
+    }
+    demo (subject) {
+        // console.log(`Today we are learing about ${subject}.`);
+        return `Today we are learning about ` + subject + `.`;
+    }
+    grade (student, subject) {
+        // console.log(student + ` recieves a perfect score on ` + subject);
+        // return student + ` receives a perfect score on ` + subject + `.`;
+        return `${student.name} receives a perfect score on ${subject}.`;
+    }
+    randomizedMath (student) {
+        // return student.grade + 5;
+        return `This method randomly changes ${student.name}'s grade of ${student.grade}.`
+        // I'll work on this more later (after I spend some time on the coding challenge)
+    }
+}
+
+class Student extends Person {
+    constructor(studentProps) {
+        super(studentProps);
+        this.previousBackground = studentProps.previousBackground;
+        this.className = studentProps.className;
+        this.favSubjects = studentProps.favSubjects;
+        this.grade = studentProps.grade;
+    }
+    listsSubjects () {
+        return `${this.favSubjects};`
```

Comment:
This will log out the whole `this.favSubjects` array. We want to log out each one, one at a time. This would be a great use of the `.forEach` method! (This is not a big deal, I'm being nitpicky here.)